src/entities/mellowLpVault.ts

```diff
@@ -0,0 +1,437 @@
+      return;
+    }
+
+    if (!this.vaultInitialized) {
+      console.log('The vault should be initialized first');
+      return;
+    }
+
+    if (isUndefined(this.readOnlyContracts)) {
+      throw new Error('Uninitialized contracts.');
+    }
+
+    this.userAddress = await this.signer.getAddress();
+    console.log('user address', this.userAddress);
+
+    this.writeContracts = {
+      token: new ethers.Contract(
+        this.readOnlyContracts.token.address,
+        IERC20MinimalABI,
+        this.signer,
+      ),
+      voltzVault: new ethers.Contract(this.voltzVaultAddress, VoltzVaultABI, this.signer),
+      erc20RootVault: new ethers.Contract(
+        this.erc20RootVaultAddress,
+        Erc20RootVaultABI,
+        this.signer,
+      ),
+    };
+
+    console.log('write contracts ready');
+
+    await this.refreshUserDeposit();
+    console.log('user deposit refreshed', this.userDeposit);
+    await this.refreshUserExpectedCashflow();
+    console.log('user expected cashflow refreshed', this.userExpectedCashflow);
+    await this.refreshWalletBalance();
+    console.log('user wallet balance refreshed', this.userWalletBalance);
+
+    this.userInitialized = true;
+  };
+
+  public get tokenName(): string {
+    if (isUndefined(this.readOnlyContracts)) {
+      return '-';
+    }
+
+    return getTokenInfo(this.readOnlyContracts.token.address).name;
+  }
+
+  public get tokenDecimals(): number {
+    if (isUndefined(this.readOnlyContracts)) {
+      return 18;
+    }
+
+    return getTokenInfo(this.readOnlyContracts.token.address).decimals;
+  }
+
+  public get protocol(): string {
+    if (isUndefined(this.protocolId)) {
+      return '-';
+    }
+
+    const prefix = getProtocolPrefix(this.protocolId);
+
+    return `${prefix}${this.tokenName}`;
+  }
+
+  refreshVaultAccumulative = async (): Promise<void> => {
+    if (isUndefined(this.readOnlyContracts)) {
+      this.vaultAccumulative = 0;
+      this.vaultCap = 0;
+      return;
+    }
+
+    const totalLpTokens = await this.readOnlyContracts.erc20RootVault.totalSupply();
+
+    if (totalLpTokens.eq(0)) {
+      this.vaultAccumulative = 0;
+      this.vaultCap = 0;
+      return;
+    }
+
+    const tvl = await this.readOnlyContracts.erc20RootVault.tvl();
+    console.log('accumulated (tvl):', tvl.minTokenAmounts[0].toString());
+
+    const nft = await this.readOnlyContracts.erc20RootVault.nft();
+    const strategyParams = await this.readOnlyContracts.erc20RootVaultGovernance.strategyParams(
+      nft,
+    );
+    console.log('strategy params:', strategyParams);
+    console.log('token limit', strategyParams.tokenLimit.toString());
+
+    this.vaultAccumulative = this.descale(tvl.minTokenAmounts[0], this.tokenDecimals);
+    this.vaultCap = this.descale(
+      totalLpTokens.mul(toBn('1', 18)).div(strategyParams.tokenLimit),
+      16,
+    );
+  };
+
+  refreshVaultExpectedApy = async (): Promise<void> => {
+    this.vaultExpectedApy = 7;
```

Comment:
do we still plan to use this in the future? if not, shall we remove existing code related to 'vault expected apy'?